dumper-integration-tests/redshift-tests/src/test/java/com/google/edwmigration/dumper/integration/SvvColumnsTest.java

```diff
@@ -0,0 +1,88 @@
+import static com.google.edwmigration.dumper.base.TestBase.CSV_PARSER;
+import static com.google.edwmigration.dumper.base.TestBase.assertDbCsvDataEqual;
+import static com.google.edwmigration.dumper.base.TestConstants.EXPORTED_FILES_BASE_PATH;
+import static com.google.edwmigration.dumper.base.TestConstants.PASSWORD_DB;
+import static com.google.edwmigration.dumper.base.TestConstants.SQL_REQUESTS_BASE_PATH;
+import static com.google.edwmigration.dumper.base.TestConstants.URL_DB;
+import static com.google.edwmigration.dumper.base.TestConstants.USERNAME_DB;
+import static com.google.edwmigration.dumper.jdbc.JdbcUtil.getDbColumnNames;
+import static com.google.edwmigration.dumper.jdbc.JdbcUtil.getStringNotNull;
+import static com.google.edwmigration.dumper.sql.SqlUtil.getSql;
+import static java.sql.DriverManager.getConnection;
+
+import com.google.common.collect.LinkedHashMultiset;
+import com.opencsv.CSVReaderHeaderAware;
+import com.opencsv.CSVReaderHeaderAwareBuilder;
+import com.opencsv.exceptions.CsvException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import org.testng.annotations.Test;
+
+public class SvvColumnsTest {
+
+  private static final String SQL_PATH = SQL_REQUESTS_BASE_PATH + "svv_columns.sql";
+  private static final String CSV_FILE_PATH = EXPORTED_FILES_BASE_PATH + "svv_columns.csv";
+
+  @Test
+  public void svvColumnsTest() throws SQLException, IOException, CsvException {
+    LinkedHashMultiset<Map<String, String>> dbMultiset = create();
+    LinkedHashMultiset<Map<String, String>> csvMultiset = create();
+    LinkedHashMultiset<String> dbColumnHeaders, csvColumnHeaders;
+
+    try (Connection connection = getConnection(URL_DB, USERNAME_DB, PASSWORD_DB);
+        PreparedStatement preparedStatement = connection.prepareStatement(getSql(SQL_PATH))) {
+      ResultSet rs = preparedStatement.executeQuery();
+
+      dbColumnHeaders = create(getDbColumnNames(rs.getMetaData()));
+
+      while (rs.next()) {
+        Map<String, String> dbRow = new HashMap<>();
+        for (String headed : dbColumnHeaders) {
+          dbRow.put(headed, getStringNotNull(rs, headed));
+        }
+        dbMultiset.add(dbRow);
+      }
+    }
+
+    try (FileReader fileReader = new FileReader(CSV_FILE_PATH);
+        CSVReaderHeaderAware reader =
+            new CSVReaderHeaderAwareBuilder(fileReader).withCSVParser(CSV_PARSER).build()) {
+      Map<String, String> csvRow;
+      while ((csvRow = reader.readMap()) != null) {
+        csvMultiset.add(csvRow);
+      }
+    }
+
+    csvColumnHeaders = create(getFirst(dbMultiset.elementSet(), null).keySet());
+
+    assertThat(dbColumnHeaders.size()).isEqualTo(csvColumnHeaders.size());
+    assertThat(dbColumnHeaders).containsExactlyElementsIn(csvColumnHeaders);
+    assertDbCsvDataEqual(dbMultiset, csvMultiset);
```

Comment:
I was wondering if this is equal to

```
    Truth.assertThat(Multisets.difference(dbMultiset, csvMultiset))
            .containsExactlyElementsIn(Multisets.difference(csvMultiset, dbMultiset));
```

Reply:
Ok, let's make an experiment.
There are 4 possible cases for this assert: 
1. A and B are equal
2. A has plus or minus some entries
3. B has plus or minus some entries
4. A and B have mutually exclusive entries

Now, with this output can you clearly understand what happened in the test?
```
value of: multiset.onlyElement()
  expected: {numeric_precision_radix=, is_nullable=NO, column_default=, numeric_scale=, table_catalog=dev, table_name=pg_tablespace, collation_catalog=, domain_name=, character_set_name=, interval_type=, datetime_precision=, collation_schema=, column_name=spcname, character_maximum_length=, collation_name=case_sensitive, character_set_catalog=, interval_precision=, character_set_schema=, ordinal_position=1, table_schema=pg_catalog, data_type=name, numeric_precision=, remarks=}
  but was : {numeric_precision_radix=, datetime_precision=, is_nullable=YES, column_default=, collation_schema=, column_name=as_locator, character_maximum_length=, collation_name=, numeric_scale=, character_set_catalog=, table_catalog=dev, table_name=routines, interval_precision=, character_set_schema=, collation_catalog=, ordinal_position=54, domain_name=character_data, character_set_name=, table_schema=information_schema, data_type=character varying, numeric_precision=, interval_type=, remarks=}
```

and can you with this
```
DB view and CSV file have mutually exclusive row(s)
DB view has 1 different row(s): 
{numeric_precision_radix=, datetime_precision=, is_nullable=YES, column_default=, collation_schema=, column_name=as_locator, character_maximum_length=, collation_name=, numeric_scale=, character_set_catalog=, table_catalog=dev, table_name=routines, interval_precision=, character_set_schema=, collation_catalog=, ordinal_position=54, domain_name=character_data, character_set_name=, table_schema=information_schema, data_type=character varying, numeric_precision=, interval_type=, remarks=}
CSV file has 1 different row(s): 
{numeric_precision_radix=, is_nullable=NO, column_default=, numeric_scale=, table_catalog=dev, table_name=pg_tablespace, collation_catalog=, domain_name=, character_set_name=, interval_type=, datetime_precision=, collation_schema=, column_name=spcname, character_maximum_length=, collation_name=case_sensitive, character_set_catalog=, interval_precision=, character_set_schema=, ordinal_position=1, table_schema=pg_catalog, data_type=name, numeric_precision=, remarks=}
```

**Conclusion**
Does the usage of this built-in comparison method provide any benefit for tests?
I don't think so. 
We save 20 lines of code, sure, but we lose a ton of output verbosity and visibility.
I completely cannot understand why we are fighting so hard for the 20-line code fragment and willing to sacrifice output readability over the number of code lines.

Reply:
The following considerations apply:
- widely used libraries are less likely to have bugs
- same applies for performance reasons
- easier for new contributors to understand what's going on (under the assumption that they know the libraries in use)
- the output format `expected: {} \n but was: {}` is very common in test failure reports (some tools might even look for this pattern)

Reply:
* libs are more likely to have bugs compare to Java native methods
* I tried both methods, this custom and provided by you above - no performance difference on 20k entries
* agree, should be easier. The problem in this case is - easier doesn't mean better. As Shevek said, we develop code that does what we need, it solves some computations or problems, here is the example of solving a problem with understand-ability of long entries log output
* again, I agree that it's very common, but we are talking about a different code property. Common vs Convenient-and-Readable I'd say it. A test must do what its supposed to do - verify and provide a report that's easy to read.

@paolomorandini Still, from the first log output above, can you recognize and distinguish - does either DB or CSV have more or less entries? or both?

Reply:
So `Truth.assertThat(dbMultiset).containsExactlyElementsIn(csvMultiset)` has been excluded because of too much output being produced (it would've been possible to solve this with an override if `Multiset.toString` wasn't final).

Should `assertDbCsvDataEqual` be refactored to use `Multisets.difference(a, b)` instead of manually removing elements with `b.forEach(a::remove)`?

Reply:
Yes, the reduction itself can be refactored.
Although, keep in mind that `Multisets.difference()` is marked as a Beta method.
```
@Beta
  public static <E extends @Nullable Object> Multiset<E> difference() {
```
which stands for:

> Signifies that a public API (public class, method or field) **is subject to incompatible changes**, **or even removal**, in a future release. An API bearing this annotation is exempt from any compatibility guarantees made by its containing library. Note that the presence of this annotation implies nothing about the quality or performance of the API in question, only the fact that it is not "API-frozen."
> It is generally safe for applications to depend on beta APIs, **at the cost of some extra work during upgrades**. However it is generally inadvisable for libraries (which get included on users' CLASSPATHs, outside the library developers' control) to do so.


@paolomorandini , I can't understand why we implement the beta functionality of the library which is a subject to at least for some alterations. It doesn't look like a step that adds stability and maintainability to our framework.
However, I see, that there is a strong opinion on using libraries, so, I'll implement corresponding changes but I've already highlighted the possible risks as well.

Done.